backend/api/go_http_client/run_client/run_service/run_service_client.go

```diff
@@ -212,6 +212,35 @@ func (a *Client) ReportRunMetrics(params *ReportRunMetricsParams, authInfo runti
 
 }
 
+/*
+TerminateRun terminate run API
+*/
+func (a *Client) TerminateRun(params *TerminateRunParams, authInfo runtime.ClientAuthInfoWriter) (*TerminateRunOK, error) {
```

Comment:
For each API that you call, please have an abstract client with a fake implementation (for tests) and a real implementation (for production). 

See this example for the persistence agent: 
https://github.com/kubeflow/pipelines/tree/master/backend/src/agent/persistence/client

Reply:
`workflow_fake.go` has been updated to fake this behavior.
see: https://github.com/kubeflow/pipelines/pull/528/files#diff-b43f69f9e25073a72d6ebec688536c24

Reply:
Thank you Riley. Looks like this PR is still WIP according to the title. Let me know when it is ready for review. Thanks!